src/compiler/checker.ts

```diff
@@ -36898,6 +36898,11 @@ namespace ts {
                     const someBaseTypeHasBothIndexers = forEach(getBaseTypes(<InterfaceType>type), base => getIndexTypeOfType(base, IndexKind.String) && getIndexTypeOfType(base, IndexKind.Number));
                     errorNode = someBaseTypeHasBothIndexers || !type.symbol.declarations ? undefined : type.symbol.declarations[0];
                 }
+                if (!errorNode) {

```

Comment:
Seems like this isn't the right fix - you'll still end up with the wrong error span, just as in the following example:

```ts
class Foo {
  [p: string]: any;
  static [p: string]: number;
}
```

```
error: Property 'prototype' of type 'Foo' is not assignable to string index type 'number'.
```

Reply:
I've opened up #44134 to track that.